service/history/mutableStateBuilder.go

```diff
@@ -1324,7 +1337,7 @@ func (e *mutableStateBuilder) addWorkflowExecutionStartedEventForContinueAsNew(d
 	event := e.hBuilder.AddWorkflowExecutionStartedEvent(req, &previousExecutionInfo.RunID)
 	e.ReplicateWorkflowExecutionStartedEvent(domainID, parentDomainID, execution, createRequest.GetRequestId(),
 		event.WorkflowExecutionStartedEventAttributes)
-
+	e.writeEventToCache(event)
```

Comment:
let's move this to ReplicateWorkflowExecutionStartedEvent as discussed

Reply:
Done